indexing-service/src/main/java/org/apache/druid/indexing/seekablestream/SeekableStreamTuningConfig.java

```diff
@@ -0,0 +1,315 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package org.apache.druid.indexing.seekablestream;
+
+import com.fasterxml.jackson.annotation.JsonCreator;
+import com.fasterxml.jackson.annotation.JsonProperty;
+import org.apache.druid.segment.IndexSpec;
+import org.apache.druid.segment.indexing.RealtimeTuningConfig;
+import org.apache.druid.segment.indexing.TuningConfig;
+import org.apache.druid.segment.realtime.appenderator.AppenderatorConfig;
+import org.apache.druid.segment.writeout.SegmentWriteOutMediumFactory;
+import org.joda.time.Period;
+
+import javax.annotation.Nullable;
+import java.io.File;
+import java.util.Objects;
+
+public abstract class SeekableStreamTuningConfig implements TuningConfig, AppenderatorConfig
+{
+  private static final int DEFAULT_MAX_ROWS_PER_SEGMENT = 5_000_000;
+  private static final boolean DEFAULT_RESET_OFFSET_AUTOMATICALLY = false;
+  private static final boolean DEFAULT_SKIP_SEQUENCE_NUMBER_AVAILABILITY_CHECK = false;
+
+  private final int maxRowsInMemory;
+  private final long maxBytesInMemory;
+  private final int maxRowsPerSegment;
+  @Nullable
+  private final Long maxTotalRows;
+  private final Period intermediatePersistPeriod;
+  private final File basePersistDirectory;
+  @Deprecated
+  private final int maxPendingPersists;
+  private final IndexSpec indexSpec;
+  private final boolean reportParseExceptions;
+  private final long handoffConditionTimeout;
+  private final boolean resetOffsetAutomatically;
+  @Nullable
+  private final SegmentWriteOutMediumFactory segmentWriteOutMediumFactory;
+  private final Period intermediateHandoffPeriod;
+  private final boolean skipSequenceNumberAvailabilityCheck;
```

Comment:
this property is used in `SeekableStreamIndexTaskRunner`, which only holds a reference to `SeekableStreamTuningConfig` and does not have access to `KinesisTuningConfig`. So although it is true that this is only used in Kinesis, I think moving this to the Kinesis config would actually make the task runner code more complicated. Thoughts?

Reply:
Hmm, I think in that case, `possiblyResetDataSourceMetadata` should be implemented differently in `KinesisIndexTaskRunner` and `KafkaIndexTaskRunner` because it's easier to understand the code by limiting the scope to check.

Reply:
okay, will do that